### ISSUE TITLE
header referer not change document.referrer

### ISSUE THREAD

**Issue by mhf-ir** [puppeteer/puppeteer#2831]
```js
// enable request interception
await page.setRequestInterception(true);
// add header for the navigation requests
page.on('request', request => {
  // Add a new header for navigation request.
  const headers = request.headers();
  headers['Referrer'] = 'https://sample.tld';
  request.continue({ headers });
});
// navigate to the website
await page.goto('https://example.com');
```
on page `console.log(documen.referrer)` will be null

**Comment by mboeser** [comment]
appears there's a typo in the console.log(_document.referrer_). could it be simply that? if not, do you have a live URL we can test?

**Comment by pomerantsev** [comment]
@mhf-ir here's what I've been able to uncover.

First off, the header name is `Referer`, not `Referrer` (yes, it's [spelled incorrectly](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Referer)). The DOM API uses the canonical spelling, hence `document.referrer`.

Other than that, there seem to be no issues with your code, and Puppeteer correctly sets the request header. You can see it yourself if you get the devtools open initially and add a small delay to all interactions with the devtools protocol: `const browser = await puppeteer.launch({devtools: true, slowMo: 100});`

However, it looks like `document.referrer` is not exactly in sync with the `Referer` header. So if you visit `https://en.wikipedia.org/wiki/Main_Page` while setting `Referer` to `https://en.wikipedia.org/wiki/Puppeteer` using Puppeteer, your `document.referrer` value will be an empty string. However, if you actually click the link from the latter page to the former, then `document.referrer` will have the expected value.

Long story short, this doesn't look like a Puppeteer issue.
